src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1360,12 +1394,12 @@ public override void Cancel()
                 return;
 
             var connection = Connection;
-            if (connection is null)
+            if (connection is null && _connector is null)
```

Comment:
nit: can just do `var connector = connection?.Connector ?? _connector` here and check that, simplifying the code below.